### ISSUE TITLE
[no-throw-literal] False positive with type assertions

### ISSUE THREAD

**Issue by cherryblossom000** [typescript-eslint/typescript-eslint#2350]
**Repro**

```json
{
  "rules": {
    "@typescript-eslint/no-throw-literal": 2
  }
}
```

```ts
declare const e: Error | string
throw e as Error
```

**Expected Result**
No error because `e` is asserted to be an `Error`.

**Actual Result**
```
2:7  error  Expected an error object to be thrown  @typescript-eslint/no-throw-literal
```

**Additional Info**
\-

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.1` |
| `@typescript-eslint/parser`        | `3.7.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.6.0` |
| `node`                             | `12.18.2` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
I believe that this is just because the rule doesn't have proper handling for all node types here

https://github.com/typescript-eslint/typescript-eslint/blob/89851fab93b7bf954f795f3ec35cd35e98d2031e/packages/eslint-plugin/src/rules/no-throw-literal.ts#L64-L95
